### PR TITLE
[libc] Add is_fixed_point type trait.

### PR DESCRIPTION
**Pull request by lntue**
This PR depends on https://github.com/llvm/llvm-project/pull/81255.

### CODE REVIEW

File: libc/src/__support/CPP/type_traits/is_fixed_point.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#include "include/llvm-libc-macros/stdfix-macros.h"
+
+namespace LIBC_NAMESPACE::cpp {
+
+// is_fixed_point
+#ifdef LIBC_COMPILER_HAS_FIXED_POINT
+template <typename T> struct is_fixed_point {
+private:
+  template <typename Head, typename... Args>
+  LIBC_INLINE_VAR static constexpr bool __is_unqualified_any_of() {
```
Comment (gchatelet) [inline libc/src/__support/CPP/type_traits/is_fixed_point.h:24]: `LIBC_INLINE` instead of `LIBC_INLINE_VAR`

**Review by gchatelet — COMMENTED** [review]
It's missing type_traits Bazel support

**Comment by michaelrj-google** [comment]
this seems to need a rebase now that https://github.com/llvm/llvm-project/pull/81255 has landed, I will review once that's done.

File: libc/src/__support/CPP/type_traits/is_fixed_point.h (hunk 1 above)
Comment (lntue) [reply libc/src/__support/CPP/type_traits/is_fixed_point.h]: Done.

File: libc/include/llvm-libc-macros/stdfix-macros.h [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __LLVM_LIBC_MACROS_STDFIX_MACROS_H
+#define __LLVM_LIBC_MACROS_STDFIX_MACROS_H
+
+#ifdef __clang__
+#if (!defined(__cplusplus) || (__clang_major__ >= 18))
+// _Fract and _Accum types are avaiable
```
[the remark this answers is not in the document]
Comment (lntue) [reply libc/include/llvm-libc-macros/stdfix-macros.h]: Done.

**Comment by lntue** [comment]
> this seems to need a rebase now that #81255 has landed, I will review once that's done.

I've synced this with HEAD.

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Comment by nickdesaulniers** [comment]
This is causing a failure in post submit

```
/home/sivachandra/libc-x86_64-debian/libc-x86_64-debian-dbg-runtimes-build/llvm-project/libc/src/__support/CPP/type_traits/is_fixed_point.h:30:23: error: use of undeclared identifier '_Fract'
   30 |       T, short fract, fract, long fract, unsigned short fract, unsigned fract,
      |                       ^
/home/sivachandra/libc-x86_64-debian/libc-x86_64-debian-dbg-runtimes-build/llvm-project/libc/include/llvm-libc-macros/stdfix-macros.h:21:15: note: expanded from macro 'fract'
   21 | #define fract _Fract
      |               ^
```
PTAL

**Comment by lntue** [comment]
> This is causing a failure in post submit
> 
> ```
> /home/sivachandra/libc-x86_64-debian/libc-x86_64-debian-dbg-runtimes-build/llvm-project/libc/src/__support/CPP/type_traits/is_fixed_point.h:30:23: error: use of undeclared identifier '_Fract'
>    30 |       T, short fract, fract, long fract, unsigned short fract, unsigned fract,
>       |                       ^
> /home/sivachandra/libc-x86_64-debian/libc-x86_64-debian-dbg-runtimes-build/llvm-project/libc/include/llvm-libc-macros/stdfix-macros.h:21:15: note: expanded from macro 'fract'
>    21 | #define fract _Fract
>       |               ^
> ```
> 
> PTAL
> 
> https://lab.llvm.org/buildbot/#/builders/225/builds/31148

It should be fixed with https://github.com/llvm/llvm-project/pull/81788

**Comment by jplehr** [comment]
This seems to have caused the lib AMD GPU buildbot to break:
https://lab.llvm.org/staging/#/builders/11/builds/2190 and the mentioned #81788 did not fix it for the AMD bot.
@jhuber6 (or someone else) can you please take a look at this?
Thanks!

**Comment by lntue** [comment]
> This seems to have caused the lib AMD GPU buildbot to break:
> https://lab.llvm.org/staging/#/builders/11/builds/2190 and the mentioned #81788 did not fix it for the AMD bot.
> @jhuber6 (or someone else) can you please take a look at this?
> Thanks!

@jplehr can you check if #81820 fixes the issue?  Thanks

**Comment by jplehr** [comment]
Unfortunately, I'm still seeing the issue in a local build with that patch applied.
However, I had to modify two things that I'll comment on the PR.

**Pull request merged**